tarofreighter/chain_porter.go

```diff
@@ -492,79 +492,41 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 }
 
 // advanceState advances the state machine.
-func (p *ChainPorter) advanceState(currentPkg *sendPackage,
+func (p *ChainPorter) advanceState(pkg *sendPackage,
 	txBroadcastRespChan chan *PendingParcel) error {
 
-	logState := func(sendState SendState) {
-		log.Infof("ChainPorter executing state: %v", sendState)
-	}
-
-	// Advance through sync evaluated states. (Up to, but not including,
-	// SendStateWaitTxConf.)
-	continueStateProgression := currentPkg.SendState < SendStateWaitTxConf
-	for continueStateProgression {
-		logState(currentPkg.SendState)
-
-		// Before we attempt a state transition, make sure that we
-		// aren't trying to shut down.
-		select {
-		case <-p.Quit:
-			return fmt.Errorf("porter shutting down")
-
-		default:
-		}
-
-		updatedPkg, err := p.stateStep(*currentPkg, txBroadcastRespChan)
-		if err != nil {
-			return err
-		}
-
-		// Continue state progression whilst target state has not yet
-		// been reached. Stop before evaluating target state.
-		continueStateProgression = updatedPkg.SendState < SendStateWaitTxConf
-
-		currentPkg = updatedPkg
-	}
-
-	// The remaining state transitions are evaluated using a goroutine
-	// because tx confirmation and proof transfer could take a while.
+	// Some states require a goroutine because tx confirmation and proof
+	// transfer can take a while.
```

Comment:
Cool I made this comment earlier, the final thing I think is to then extract this out into the state machine that just needs to wait for completion and do the proof transfer, etc.

Reply:
As is, we aren't really set up for concurrent funding, since we don't have a sort of UTXO leasing or coin selection lock like we do in `lnd`.